blobstore/upload.go

```diff
@@ -122,6 +124,27 @@ func (bh *BlobHandler) HandleMultipartUpload(c echo.Context) error {
 	}
 
 	bucket := c.QueryParam("bucket")
+	if bh.Config.AuthLevel > 0 {
+		claims, ok := c.Get("claims").(*auth.Claims)
+		if !ok {
+			return c.JSON(http.StatusInternalServerError, "Could not get claims from request context")
+		}
+		roles := claims.RealmAccess["roles"]
+		ue := claims.Email
+
+		// Check for required roles
+		isAdmin := utils.StringInSlice(bh.Config.AdminRoleName, roles)
```

Comment:
We want to check if the users are admins or superwriter because if they are then them being limited_writer does not limit them.

In your proposed implementation what if someone is admin but also limited_writer (hypothetical, but we should handle this edge case), they would be limited despite being admins.

Reply:
It can happen when someone were a limited_writer and they were elevated to super_writer/writer or admin, their previous role was not removed. Current design provides fault tolerance for this, suggested design would break.

I would suggest keeping it like how it is right now since this provides an added level of fault tolerance and a hierarchy. It also separates the role based auth from FGAC.

Let me know if you still think this needs to be edited.

Reply:
Users are not directly assigned roles, but instead are added to groups that have assigned roles. A group would not have limited_writer and admin role attached, that does not make sense. While I understand what you are saying, I think it does not fit in with the authorization schema we have developed and adds extra complexity that is unneeded. If we were attaching roles to individual users then I would agree that maybe we need to handle that edge case, but the reason for FGAC is to in some ways avoid doing that

It also adds envs and extra lines of code to each handler where you want to implement FGAC

Reply:
Ok.